src/scaffoldmaker/meshtypes/meshtype_3d_bladder1.py

```diff
@@ -0,0 +1,770 @@
+
+    @classmethod
+    def getOptionScaffoldTypeParameterSetNames(cls, optionName, scaffoldType):
+        if optionName == 'Ureter':
+            return list(cls.ostiumDefaultScaffoldPackages.keys())
+        assert scaffoldType in cls.getOptionValidScaffoldTypes(optionName), cls.__name__ + '.getOptionScaffoldTypeParameterSetNames.  ' + \
+            'Invalid option \'' + optionName + '\' scaffold type ' + scaffoldType.getName()
+        return scaffoldType.getParameterSetNames()
+
+    @classmethod
+    def getOptionScaffoldPackage(cls, optionName, scaffoldType, parameterSetName=None):
+        '''
+        :param parameterSetName:  Name of valid parameter set for option Scaffold, or None for default.
+        :return: ScaffoldPackage.
+        '''
+        if parameterSetName:
+            assert parameterSetName in cls.getOptionScaffoldTypeParameterSetNames(optionName, scaffoldType), \
+                'Invalid parameter set ' + str(parameterSetName) + ' for scaffold ' + str(scaffoldType.getName()) + ' in option ' + str(optionName) + ' of scaffold ' + cls.getName()
+        if optionName == 'Ureter':
+            if not parameterSetName:
+                parameterSetName = list(cls.ostiumDefaultScaffoldPackages.keys())[0]
+            return copy.deepcopy(cls.ostiumDefaultScaffoldPackages[parameterSetName])
+        assert False, cls.__name__ + '.getOptionScaffoldPackage:  Option ' + optionName + ' is not a scaffold'
+
+    @classmethod
+    def checkOptions(cls, options):
+        if not options['Ureter'].getScaffoldType() in cls.getOptionValidScaffoldTypes('Ureter'):
+            options['Ureter'] = cls.getOptionScaffoldPackage('Ureter', MeshType_3d_ostium1)
+        if (options['Number of elements up 1'] < 4):
+            options['Number of elements up 1'] = 4
+        if (options['Number of elements up 2'] < 4):
+            options['Number of elements up 2'] = 4
+        if (options['Number of elements around'] < 8):
+            options['Number of elements around'] = 8
+        if (options['Number of elements through wall'] != 1):
+            options['Number of elements through wall'] = 1
+        if (options['Major diameter'] < options['Minor diameter']):
+            options['Major diameter'] = options['Minor diameter']
+
+    @staticmethod
+    def generateBaseMesh(region, options):
+        '''
+        Generate the base tricubic Hermite mesh. See also generateMesh().
+        :param region: Zinc region to define model in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: None
+        '''
+
+        elementsCountUp1 = options['Number of elements up 1']
+        elementsCountUp2 = options['Number of elements up 2']
+        elementsCountAround = options['Number of elements around']
+        height = options['Height']
+        majorDiameter = options['Major diameter']
+        minorDiameter = options['Minor diameter']
+        radius = options['Urethra radius']
```

Comment:
Standardise to use diameter in user options - multiply by 0.5 to get radius variable.